Source/OpenPF2Core/Private/PF2CharacterBase.cpp

```diff
@@ -237,3 +261,13 @@ void APF2CharacterBase::ClearManagedPassiveGameplayEffects()
 
 	this->bManagedPassiveEffectsGenerated = false;
 }
+
+void APF2CharacterBase::HandleCharacterLevelChanged(const float OldLevel, const float NewLevel)
+{
+	this->DeactivatePassiveGameplayEffects();
+
+	this->CharacterLevel = NewLevel;
+	this->OnCharacterLevelChanged(OldLevel, NewLevel);
+
+	this->ActivatePassiveGameplayEffects();
```

Comment:
Addressed by introducing `RemoveRedundantPendingAbilityBoosts()` which now gets called as part of `ActivatePassiveGameplayEffects()` (both methods are at _player character level_) to ensure that boost GAs are removed if they've already received a selection. This only happens if the game designer goes through the PF2CharacterInterface rather than toggling passive GEs via the ASC.